detectors/node/opentelemetry-resource-detector-container/package.json

```diff
@@ -51,6 +51,7 @@
     "@opentelemetry/api": "^1.0.0"
   },
   "dependencies": {
+    "@kubernetes/client-node": "^0.20.0",
```

Comment:
A concern here is that this is a large dependency, but mostly that it still uses the deprecated `request` HTTP client library. This would bring in some CVEs (https://github.com/kubernetes-client/javascript/issues/754#issuecomment-2003282886).

There is an effort (https://github.com/kubernetes-client/javascript/blob/master/FETCH_MIGRATION.md https://github.com/kubernetes-client/javascript/issues/754) to move away from `request` to `fetch()`. That looks active. There was a recent release-candidate release with that work: `'1.0.0-rc4': '2023-12-23T15:06:31.075Z'`. However, I haven't looked deeply and I'm not sure how soon that might come.

Even if that came, an issue with `@kubernetes/client-node@1.x` might be that I believe is dropped compat with Node.js v14 in https://github.com/kubernetes-client/javascript/commit/3a995fbb8399dfb9d6751006ed9f2deb5c03c7a2
That commit unfortunately doesn't mention the Node.js version, nor is it tied to a PR with any discussion.
That same commit is dropping an AbortController shim for older Node.js version support. AbortController *is* avaialble starting in Node.js v14.17.0, so it is *possible* that  `@kubernetes/client-node@1.x` would work with `node >=14.17.0`. They don't test with Node v14 though, so that is tenuous support.

I haven't read through this PR yet. Would it be possible to do whatever k8s client calls **without using `@kubernetes/client-node`**? I don't know if it is a very complex API to call manually.

Another option would possibly be to wait for `@kubernetes/client-node@1.x` **and** the OTel SDK 2.0 milestone (https://github.com/open-telemetry/opentelemetry-js/milestone/17), because that will drop support for Node.js v14.

Reply:
I believe - This api takes care of adding filepath required for certifcates etc to be present. 
I think we can certainly manually do all the operations that this does. It may be bit more code . I can look into it. 

I see the migration to fetch and partial compatibility for node 14 as well. I think officially node 14 is not supported now. 

Let me take a look at the effort.

Reply:
I was looking  at the library itself. 

It is officially maintained by [Kubernetes SIG API Machinery](https://github.com/kubernetes/community/tree/master/sig-api-machinery).

I suggest we use this and create an issue there regarding node 14 and request --> fetch ( which is I believe is already in progress )

@trentm let me know what you think.